call-a-function-api-oci-clouddev.md

```diff
@@ -0,0 +1,772 @@
+5. Deploy the function. 
+
+    ```console
+    fn -v deploy --app your-app-name
+    ```
+
+    Various messages are displayed as the Docker images are built, pushed to OCIR, and eventually deployed to Oracle Functions.
+
+6. Invoke the function. 
+
+    ```console
+    fn invoke your-app-name my-func-name
+    ```
+
+    Returns: `{"message": "Hello World"}`
+
+7. Invoke the function with a parameter. 
+
+    ```console
+    echo -n '{"name":"Bob"}' | fn invoke your-app-name my-func-name
+    ```
+
+    Returns: `{"message": "Hello Bob"}`
+
+## 3. Create an API Gateway
+
+To call your function, create an API Gateway.
+
+### Create the API Gateway
+
+1. Open the navigation menu and click **Developer Services**. Under **API Management**, click **Gateways**.
+2. Select your compartment from the **Compartment** drop-down.
+3. Click **Create Gateway**
+4. Fill in the following information to define your API Gateway. 
+    * **Name:** `<your-gateway-name>`
+    * **Type:** `<Public>`
+    * **Compartment:** `<your-compartment-name>`
+    * **Virtual Cloud Network in `<your-vcn-name>:`** `<select-your-vcn>`
+    * **Subnet in `<your-compartment-name>:`** `<your-public-subnet-name>`
+5. Click **Create**. Wait a few minutes for your API Gateway to be created.
+
+### Create an API Deployment for your Gateway
+
+1. Click **Deployments** in Resources section on the left side of the screen.
+2. Click **Create Deployment**.
+3. Ensure that **From Scratch** is selected for the deployment type.
+4. To define your deployment, fill in the **Basic Information** section.
+    * **Name:** `<your-deployment-name>`
+    * **Path Prefix (example):** `/v1`
+    * **Compartment:** `<your-compartment-name>`
+    * **API Request Policies:** Take default values
+    * **API Logging Policies:** Take default value of **Information**
+5. Click **Next**. The **Routes** dialog appears with **Route 1** selected.
+6. To define your route, fill in the **Route 1** section.
+    * **Path:** `<your-route-path>` 
+        Example: /http-info
+    * **Methods:** GET POST
+    * **Type:** Oracle Functions
+    * **Application in `<your-compartment-name>`:** Select the Functions application you created.
+    * **Function Name:** Select the function you created in the configuration section.
+7. Click **Next**. The Review dialog is displayed summarizing the choices you have made.
+8. Click **Create**. Your deployment is created.
+9. Click the Deployments link for your gateway. Copy the base end point for the deployment you created. 
+
+    For example: `https://aaaaa.apigateway.us-ashburn-X.oci.customer-oic.com/v1`
+
+### Test your API Gateway
+
+With your API Gateway and deployment created, you can now test you installation. Create a simple script for the `curl` command. To create the URL for `curl`, add your deployment path to your endpoint.
+
+1. Create the script file: `touch gtw01.sh && chmod 755 gtw01.sh`
+2. Add the command curl command to the script file:
+
+    ```console
```

Comment:
bash

Reply:
Well that's just embarrassing on my part! But a cheat sheet on ID'ing what we'll be using might be good if I bonk my noggin